### PR TITLE
MINOR: Kafka Streams Scala API cleanup

### PR DESCRIPTION
**Pull request by mjsax**
- remove wildcard imports
- avoid the usage of implicit conversions
- some minor side cleanup

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @bbejeck @vvcephei 

This PR is based on this comment: https://github.com/apache/kafka/pull/7847/files#r359449354

File: streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/ImplicitConversions.scala [hunk 1]
```diff
@@ -46,12 +46,12 @@ object ImplicitConversions {
   implicit def wrapKGroupedStream[K, V](inner: KGroupedStreamJ[K, V]): KGroupedStream[K, V] =
     new KGroupedStream[K, V](inner)
 
-  implicit def wrapSessionWindowedKStream[K, V](inner: SessionWindowedKStreamJ[K, V]): SessionWindowedKStream[K, V] =
```
Comment (mjsax) [inline streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/ImplicitConversions.scala:49]: Just some reordering to keep an overview

File: streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala [hunk 2]
```diff
@@ -586,51 +597,49 @@ class KStream[K, V](val inner: KStreamJ[K, V]) {
    * @return a [[KStream]] that contains join-records for each key and values computed by the given `joiner`,
    *                 one for each matched record-pair with the same key
    * @see `org.apache.kafka.streams.kstream.KStream#leftJoin`
-   * @deprecated since 2.4. Use [[KStream#leftJoin(KStream, ValueJoiner, JoinWindows, StreamJoined)]] instead.
    */
-  @deprecated
```
Comment (mjsax) [inline streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala:591]: This method is not deprecated... (it was not release in this state luckily...)

File: streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala [hunk 3]
```diff
@@ -508,47 +519,7 @@ class KStream[K, V](val inner: KStreamJ[K, V]) {
     joiner: (V, VO) => VR,
     windows: JoinWindows
   )(implicit streamJoin: StreamJoined[K, V, VO]): KStream[K, VR] =
-    inner.join[VO, VR](otherStream.inner, joiner.asValueJoiner, windows, streamJoin)
```
Comment (mjsax) [inline streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala:511]: Just some reordering of methods... First all stream-stream, than all stream-table, than all stream-globalTable joins

**Comment by bbejeck** [comment]
Java 11/2.12 failed with `Execution failed for task ':core:integrationTest'.`

Java 11/2.13 failed with `kafka.server.epoch.EpochDrivenReplicationProtocolAcceptanceTest.shouldSurviveFastLeaderChange` ticket already exists

Java 8/2.11 fails to compile; compatibility diff between 2.11 and 2.12

File: streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KTable.scala [hunk 4]
```diff
@@ -22,15 +22,21 @@ package kstream
...
 import org.apache.kafka.streams.kstream.{ValueJoiner, ValueTransformerWithKeySupplier, KTable => KTableJ}
-import org.apache.kafka.streams.scala.FunctionsCompatConversions._
-import org.apache.kafka.streams.scala.ImplicitConversions._
+import org.apache.kafka.streams.scala.FunctionsCompatConversions.{
+  FunctionFromFunction,
+  KeyValueMapperFromFunction,
+  MapperFromFunction,
+  PredicateFromFunction,
+  ValueMapperFromFunction,
+  ValueMapperWithKeyFromFunction,
```
Comment (bbejeck) [inline streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KTable.scala]: Extra `,` at the end of the line.  Maybe this is the issue with `Scala 2.11`? Line number matches up from the `stdout` error.

**Review by bbejeck — COMMENTED** [review]
Thanks for this cleanup @mjsax, LGTM overall.  One very minor issue that hopefully resolves the `2.11` compilation issue.

**Comment by mjsax** [comment]
Updated this. Let's see if the build passes now.

**Pull request merged**